src/Algebra/Action/Structures/Raw.agda

```diff
@@ -0,0 +1,77 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Raw Actions of one (pre-)Setoid on another
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Relation.Binary.Core using (Rel)
+
+module Algebra.Action.Structures.Raw
+  {c a ℓ r} {M : Set c} {A : Set a} (_≈ᴹ_ : Rel M ℓ) (_≈_ : Rel A r)
+  where
+
+open import Data.List.Base
+  using (List; []; _∷_; _++_; foldl; foldr)
+open import Data.List.Relation.Binary.Pointwise as Pointwise
+  using (Pointwise; []; _∷_)
+open import Function.Base using (id)
+open import Level using (Level; _⊔_)
+
+private
+  variable
+    x y z : A
+    m n p : M
+    ms ns ps : List M
+
+------------------------------------------------------------------------
+-- Basic definitions: fix notation
+
+record IsRawLeftAction : Set (a ⊔ r ⊔ c ⊔ ℓ) where
+  infixr 5 _ᴹ∙ᴬ_ _ᴹ⋆ᴬ_
+
+  field
+    _ᴹ∙ᴬ_  : M → A → A
+    ∙-cong : m ≈ᴹ n → x ≈ y → (m ᴹ∙ᴬ x) ≈ (n ᴹ∙ᴬ y)
+
+-- derived form: iterated action, satisfying congruence
+
+  _ᴹ⋆ᴬ_ : List M → A → A
```

Comment:
Hmm, there's also an action of non-empty lists: there is no need for a unit to exist for this to make sense.

Reply:
A *very* good point, but I had *definitely* run out of steam for considering `Magma`/`Semigroup` actions as well, ... and their `Free` instances arising from variations on `Data.List`... but the `NonEmpty` case is one previously mentioned by @Taneb under #1281 / #2300 

But here, I think I'm strongly tempted to note this as a downstream `addition` request/suggestion, on the (oftentimes: your!) usual lines of "don't make the perfect the enemy of the good"... ;-)

Reply:
Plus... irritatingly, given the factorisation of `Data.List.NonEmpty` exhibiting ` List⁺` as a (proxy-) pair of an element and a `List`, then to define the `NonEmpty`/`Semigroup` actions, we need the `List`/`Monoid` ones! So... I decided that this should *definitely* be a downstream addition!

Reply:
So: have I left enough room for such additions, or not? 
I think that I have, but your (pertinent!) comment makes me wonder...

Reply:
So I think overall, it does allow enough room - but the naming doesn't.

Reply:
Not sure I understand: the `NonEmpty` iterated action would have a separate name, surely?

Reply:
Of course - but since `_ᴹ⋆ᴬ_` is taken, what kind of monster would be needed for that?

Reply:
I'd been thinking something like `_ᴹ⁺ᴬ_`, with the '⁺' signifiying `NonEmpty`, and the `⋆` signifying `List`...

Reply:
But: review comments are a good place to recommend 'better' names to a PR author ;-)

Reply:
See the latest commits for reorganisation and new (better?) notation...